slic3r/export2slicer.FCMacro

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Opens current visible objects in Slic3r. One can use different slicing software
+
+__Title__ = "Open in Slic3r"
+__Author__ = "Damian Łoziński"
+__Version__ = "0.1"
+__Date__ = "2019-03-06"
+__Comment__ = "Exports current visible objects to amf files and opens them in Slic3r"
+__Web__ = ""
+__Wiki__ = "http://www.freecadweb.org/wiki/index.php?title=Open in Slic3r"
+__Icon__ = "/usr/lib/freecad/Mod/plugins/icons/Slic3r.png"
+__Help__ = "Run this macro to open visible objects in slicer software. You may need to change slicer path"
+__Status__ = ""
+__Requires__ = "freecad 0.17"
+__Communication__ = "https://github.com/dlozinski"
+__Files__ = "Slic3r.png"
+
+import Mesh
+import MeshPart
+import os
+import subprocess
+
+# Slicing program executable. One can use different slicing software here and provide proper slicer executable location and custom flags
+SLICER_CMD = ["/Applications/Slic3r.app/Contents/MacOS/Slic3r", "--gui"]
+
+# You can format change to stl, but then all of the objects will be marged into single geometry
+OUTPUT_FORMAT = "amf"
+
+# Angular deflection in radians. Lower value = better quality of curved shapes and bigger file size
+ANGULAR_DEFLECTION = 0.0872665
+
+def main():
+    doc = App.ActiveDocument
+    if not doc:
+        return "No active document"
+
+    # Get all visible objests with shapes
+    visible_shapes = (o for o in doc.Objects if o.ViewObject.isVisible() and hasattr(o, 'Shape'))
+
+
+    # Ensure to not include both parent and child objects
+    objects_to_export = set()
+    for o in visible_shapes:
+        if not set(o.InListRecursive).intersection(objects_to_export):
+            objects_to_export.add(o)
+
+    if not objects_to_export:
+        return "No visible objects to export"
+
+    out_dir = os.path.dirname(doc.FileName) or "."
```

Comment:
tempfile is inconbvenient here. Stf file in the same dir  is usefull for me

Reply:
If I understand tempfile module correctly It won't work here well because I open this file in separat process asynchronously and the file might be deleted as soon as the interpreted leaves main() function body, so the Slicer won't have it

Reply:
Ok, didn't know that.